src/main/java/com/exactpro/th2/crawler/util/CrawlerUtils.java

```diff
@@ -0,0 +1,163 @@
+                                                            EventsSearchInfo info) {
+
+        EventSearchRequest.Builder eventSearchBuilder = info.searchBuilder;
+        EventSearchRequest request;
+
+        eventSearchBuilder
+                .setMetadataOnly(BoolValue.newBuilder().setValue(false).build())
+                .setStartTimestamp(info.from)
+                .setEndTimestamp(info.to)
+                .setResultCountLimit(Int32Value.of(info.batchSize));
+
+        if (info.resumeId == null)
+            request = eventSearchBuilder.build();
+        else
+            request = eventSearchBuilder.setResumeFromId(info.resumeId).build();
+
+        return function.apply(request);
+    }
+
+    public static Iterator<StreamResponse> searchMessages(Function<MessageSearchRequest, Iterator<StreamResponse>> function,
+                                                            MessagesSearchInfo info) {
+
+        MessageSearchRequest.Builder messageSearchBuilder = info.searchBuilder;
+        MessageSearchRequest request;
+
+        messageSearchBuilder
+                .setStartTimestamp(info.from)
+                .setEndTimestamp(info.to)
+                .setResultCountLimit(Int32Value.of(info.batchSize))
+                .setStream(StringList.newBuilder().addAllListString(info.aliases).build());
+
+        if (info.resumeIds == null)
+            request = messageSearchBuilder.build();
+        else
+            request = messageSearchBuilder.addAllMessageId(info.resumeIds.values()).build();
+
+        return function.apply(request);
+    }
+
+    public static List<EventData> collectEvents(Iterator<StreamResponse> iterator, Timestamp to) {
+        List<EventData> data = new ArrayList<>();
+
+        while (iterator.hasNext()) {
+            StreamResponse r = iterator.next();
+
+            if (r.hasEvent()) {
+                EventData event = r.getEvent();
+
+                if (!event.getStartTimestamp().equals(to)) {
+                    data.add(event);
+
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("Got event {}", MessageUtils.toJson(event, true));
+                    }
+                }
+            }
+        }
+
+        return data;
+    }
+
+    public static List<MessageData> collectMessages(Iterator<StreamResponse> iterator, Timestamp to) {
+        List<MessageData> messages = new ArrayList<>();
+
+        while (iterator.hasNext()) {
+            StreamResponse r = iterator.next();
+
+            if (r.hasMessage()) {
+                MessageData message = r.getMessage();
+
+                if (!message.getTimestamp().equals(to)) {
+                    messages.add(message);
+
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("Got message {}", MessageUtils.toJson(message, true));
+                    }
+                }
+            }
+        }
+
+        return messages;
+    }
+
+    public static class EventsSearchInfo {
```

Comment:
Check the required parameters in EventsSearchInfo and MessagesSearchInfo constructors for `null`